### ISSUE TITLE
Built-in RAG

### ISSUE THREAD

**Issue by richardbowman** [longy2k/obsidian-bmo-chatbot#116]
Would be awesome to embed a vector DB like something like open-webui does that indexes all of the documents in the Obsidian vault so you can talk with the model about any document.

**Comment by GJoe2** [comment]
Yes please, I have an Obsidian - Zotero integration running and something I would like is talking to my personal cite notes between different authors

**Comment by metalsp0rk** [comment]
Doesn't seem to be any traction on this one; I'm interested in this, and would be willing to take a stab at implementing something for it.  I strictly use locally-run LLMs, so I can really only test against things like Ollama, but I am willing to spend some time to get the ground-work done for this, if the Maintainer is willing to accept a PR.
